sockets/windows_socket.go

```diff
@@ -0,0 +1,18 @@
+// +build windows
+
+package sockets
+
+import (
+	"net"
+
+	"github.com/Microsoft/go-winio"
+)
+
+// NewWindowsSocket creates a Windows named pipe on the specified path.
+func NewWindowsSocket(addr string, pipeConfig *winio.PipeConfig) (net.Listener, error) {
```

Comment:
It pretty much is... Again, I added it for symmetry with other New*Socket functions. I'm not fond of the design either but I tried to adhere to it. I hear that you want to get rid of go-connections and I won't try to stop you. In any case, this changeset is mostly required for my other PR that depends on it https://github.com/docker/go-plugins-helpers/pull/63